src/classes/GE_GiftEntryController.cls

```diff
@@ -65,16 +65,18 @@ public with sharing class GE_GiftEntryController {
             GE_PaymentServices.ResponseWrapper response = paymentServices.createTransaction();
 
             return JSON.serialize(response);
-        } catch (Exception ex) {
-            Boolean isCalloutException = ex instanceof System.CalloutException;
-            Boolean hasTimedout = isCalloutException && ex.getMessage().contains(TIMED_OUT);
-
-            if (hasTimedout) {
-                return getPurchaseCallTimeoutResponse(ex, dataImportRecordId);
-            } else {
-                String JSONExceptionData = ERR_ExceptionData.createExceptionWrapperJSONString(ex);
-                throw returnDmlException(JSONExceptionData);
+        } catch (CalloutException ex) {
+            Boolean hasTimedOut = ex.getMessage().contains(TIMED_OUT);
+            String timeoutResponse = '';
+
+            if (hasTimedOut) {
+                timeoutResponse = getPurchaseCallTimeoutResponse(ex, dataImportRecordId);
             }
+
+            return timeoutResponse;
+        } catch(AuraHandledException ex) {
+            String JSONExceptionData = ERR_ExceptionData.createExceptionWrapperJSONString(ex);
+            throw returnAuraException(JSONExceptionData);
```

Comment:
I'm no fan of the original block, but to avoid the risk of causing a bug we need to revert to the original code here.

Reply:
done